File: set_in_orbit.jl
using ArgParse
using LilGuys

function set_in_orbit(snap, p, v, max_radius=nothing)
    centered = center(snap)
    if max_radius !== nothing
        r = get_r(centered)
        centered = centered[r .< max_radius]
    end
    dp = p ./ R_0 
    dv = v ./ V_0

    centered = shift_snapshot(centered, dp, dv)  # Assuming shift_snapshot is a function you have
    pf, vf = get_most_bound(centered)  # Assuming get_most_bound is a function you have

    println("center at ", pf * R_0)
    println("moving at ", vf * V_0)

    return centered
end

function parse_arguments()
    s = ArgParseSettings()
    @add_arg_table! s begin
        "input"
            arg_type=String
            help="input hdf5 file"
        "output"
            arg_type=String
            help="output hdf5 file"
        "-p", "--position"
            arg_type=Float64
            nargs='+'
            help="initial position in kpc"
        "-v", "--velocity"
            arg_type=Float64
            nargs='+'
            help="initial velocity in km/s"
        "--max_radius"
            arg_type=Float64
            default=nothing
            help="clip radius"
    end

    return parse_args(s)
end

function main()
    args = parse_arguments()
    snap = Snapshot(args["input"]) 
    new_snap = set_in_orbit(snap, args["position"], args["velocity"], args["max_radius"])
    save(new_snap, args["output"]) 
end


if abspath(PROGRAM_FILE) == @__FILE__
    main()
end

File: old_codes/set_in_orbit.py
from lilguys import Snapshot, units, profile
import argparse
import numpy as np



def get_args():
    parser = argparse.ArgumentParser()
    parser.add_argument('input', type=str,
            help="input hdf5 file")
    parser.add_argument('output', type=str,
            help="output hdf5 file")
    parser.add_argument('-p', '--position', type=float, nargs="+",
            help="initial position in kpc")
    parser.add_argument('-v', '--velocity', type=float, nargs="+",
            help="initial velocity in km/s")
    parser.add_argument('--max_radius', type=float, default=None,
            help="clip radius")

    args = parser.parse_args()
    return args

def set_in_orbit(snap, p, v, max_radius=None):
    centered = profile.center_snapshot(snap, verbose=True)
    if max_radius is not None:
        centered = centered.filter(centered.r < max_radius)
    dp = np.array(p) / units.R_0
    dv = np.array(v) / units.V_0
    
    centered.shift(dp, dv, inplace=True)
    pf, vf = profile.get_most_bound(centered)


    print("center at ", pf*units.R_0)
    print("moving at ", vf*units.V_0)

    return centered

def main():
    args = get_args()
        
    snap = Snapshot.file(args.input)
    centered = set_in_orbit(snap, args.position, args.velocity, args.max_radius)
    centered.save(args.output)

if __name__=="__main__":
    main()
